src/openassetio-core/include/openassetio/TraitsData.hpp

```diff
@@ -148,6 +148,14 @@ class OPENASSETIO_CORE_EXPORT TraitsData final {
   void setTraitProperty(const trait::TraitId& traitId, const trait::property::Key& propertyKey,
                         trait::property::Value propertyValue);
 
+  /**
+   * Returns the properties set for a given trait.
+   *
+   * If the trait has not been given to this instance, or the trait has
+   * no properties set, then it will return an empty set.
+   */
+  [[nodiscard]] trait::TraitProperties traitProperties(const trait::TraitId& traitId) const;
```

Comment:
[consider] Naming things. I still feel that `traitProperties` is a bit misleading - my first instinct would be that this returns a dictionary of properties, rather than a set of keys. I think `traitPropertyKeys` / `traitPropertyNames` / `traitPropertiesKeys` or similar would be clearer.

Reply:
Yeah, good idea. I went for singular `Property` [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/595/commits/5392327b1fe9b2923b12e041c80625a8d78980eb), as otherwise it seemed like the keys for all the properties. 🤷‍♂️